ecommerce-ui/src/ecommerce/Rental.js

```diff
@@ -0,0 +1,99 @@
+import React, { Component } from 'react';
+import './Rental.css';
+// import Cart from './ShoppingCart';
+
+class Rental extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            items: [],
+            counter: 0
+        };
+    }
+
+    addItem = ()  => {
+        this.setState((prevState, props) => {
+            const items = prevState.items;
+            const item = prevState.counter;
+            let newCount = item + 1;
+            // alert('add an item', newCount);
+            //add item to shopping cart
+            if (item) {
+                items.push(newCount);
+                // alert('One item got added', item);
+                // item++;
+            } else {
+                // alert('You should add an item');
+            }
+
+            return {
+                item: items,
+                counter: newCount
+            };
+        });
+    }
+
+    render() {
+        const {
+            houseType,
+            title,
+            image, 
+            payment,
+            location,
+            rating
+        } = this.props.rental;
+
+        const rating_stars = '*****';
+
+        return (
+            <div className='rental-property'>
+                <button onClick={this.addItem} className="btn">Add To Cart</button>
```

Comment:
It should be calling an external function that was passed into the component
this.props.addItem(title, payment)